### PR TITLE
Feature/Download Video Attachments [LEI-438]

### PR DESCRIPTION
**Pull request by pattisdr**
# Ticket
https://openscience.atlassian.net/browse/LEI-438

⭐️ Need /.aws/credentials file with AccessKey and SecretKey and .aws/config file (not added here)

# Purpose
Use boto to browse s3 for videos by study, list videos in user interface, generate 1 time tokenized time based urls for downloading the videos when the link to download videos is clicked.

# TODO
- [x] Show videos responses to a study on study attachments page
- [x] Don't show preview data collected
- [x] Search/sort video attachments
- [x]  Generate 1 time tokenized time-based url for downloading individual video
- [x] List videos related to specific response on individual response detail tab
- [x] Maybe separate responses page into three views - with the video downloading, there is a lot here - individual, all-responses, and attachments.
- [x] Add participant demographic information to study response download info
- Bulk download and enforcing viewing consent video first not included in this PR

### CODE REVIEW

File: get_study_attachments.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import sys
+import boto3
+import botocore
+
+BUCKET_NAME = 'mitLookit'
```
Comment (cwisecarver) [inline get_study_attachments.py]: This should be in a setting that pulls from an environment variable and has a default value for local development that isn't this real bucket name. We don't have to scrub this commit but defaults.py should have something fake and then developers can put a real value in their local.py that doesn't get committed.

File: studies/templates/studies/study_responses.html [hunk 2]
```diff
@@ -8,11 +8,10 @@
     <script type="application/javascript">
         $(document).ready(function(){
             hideResponses();
-            prepareDataForDownload('.all-data', 'download-all-data', 'json');
             prepareDataForDownload('#json-data-1', 'download-individual-data', 'json');
-            $(".all-csv-data").hide();
             $( ".selectable-participant" ).first().addClass('selected');
             $( ".response-data" ).first().show();
+            $( ".response-attachments" ).first().show();
```
Comment (cwisecarver) [inline studies/templates/studies/study_responses.html]: `$(".response-attachments")` why the extra spaces ? Some weird editor formatting?

File: studies/templates/studies/study_responses.html [hunk 3]
```diff
@@ -127,175 +88,101 @@
 <div class='container mb-lg'>
     <div class='row'>
         <div class="col-xs-12 col-sm-7 col-md-8 col-lg-9">
-            <a href="{% url 'exp:study-list' %}"> Manage Studies </a> &gt; <a href="{% url 'exp:study-detail' pk=study.id %}"> {{ study.name}} </a> &gt; Responses
+            <a href="{% url 'exp:study-list' %}"> Manage Studies </a> &gt; <a href="{% url 'exp:study-detail' pk=study.id %}"> {{ study.name}} </a> &gt; Individual Responses
```
Comment (cwisecarver) [inline studies/templates/studies/study_responses.html]: Really doesn't matter, but bootstrap does have a thing for [breadcrumbs](http://bootstrapdocs.com/v3.3.6/docs/components/#breadcrumbs).

**Comment by pattisdr** [comment]
@cwisecarver thanks, making changes now

File: studies/templates/studies/study_responses.html (hunk 2 above)
Comment (pattisdr) [inline studies/templates/studies/study_responses.html]: copy paste mistake

File: studies/templates/studies/study_responses.html (hunk 3 above)
Comment (pattisdr) [inline studies/templates/studies/study_responses.html]: bootstrap's are way prettier, thanks. changed.

**Comment by pattisdr** [comment]
@cwisecarver, changes made.

**Comment by cwisecarver** [comment]
@pattisdr I'm ready to merge this, but could you put some more details about where the creds files need to go and @binoculars so he can add them to the helm/secrets stuff before I merge? @icereval has us setup so that merges to develop will trigger a deploy and I'm worried what will happen if this is deployed before all the aws things are in place. @binoculars can probably work out a secure way for you to transfer the creds to him.

**Comment by cwisecarver** [comment]
Alternatively, if you want to Apple Message them to me I can airdrop them to him in the morning.

**Comment by pattisdr** [comment]
Ok I restored the bulk download items with a TODO to move to celery.

File: project/settings/defaults.py [hunk 4]
```diff
@@ -31,6 +31,10 @@
 AUTH_USER_MODEL = 'accounts.User'
 GUARDIAN_MONKEY_PATCH = False
 
+# S3 Bucket name where video attachments are stored
+BUCKET_NAME = os.environ.get('BUCKET_NAME', 'fakeBucketName')
```
Comment (cwisecarver) [inline project/settings/defaults.py]: @pattisdr Just for future reference, settings that will be different per environment should be expected to be in each environment and we should provide a default that either a) works on local dev or b) pretty clearly spells out that it's fake.

**Comment by binoculars** [comment]
@cwisecarver once https://travis-ci.org/CenterForOpenScience/helm-charts/builds/265328456 passes, you should be g2g

**Pull request merged**